### ISSUE TITLE
[BUG] Data Validation breaks Excel spreadsheet

### ISSUE THREAD

**Issue by demontiejr** [PHPOffice/PHPExcel#1028]
Hi,

I'm using Maatwebsite's Laravel Excel (package v2.1.*) and Laravel 5.1 with some native PHPExcel methods for data validation.

I'm generating an Excel spreadsheet and I want a drop down list for some columns. So, I have the following code that runs for 1000 cells in the same column. This code works fine when I open the spreadsheet using LibreOffice. However, when I open it on Excel, I get the following error: "We found a problem with some content in 'filename'. Do you want us to try to recover as much as we can?"

Here's a screenshot of the recovered spreadsheet. It doesn't work as expected and the formatting is getting weird.

![captura de tela de 2016-09-30 16_04_27](https://cloud.githubusercontent.com/assets/5882230/19004427/4eae5bb6-872c-11e6-9e38-e3ec8159ed63.png)

The code is as follows:

```php
$objValidation = $cell->getDataValidation();
$objValidation->setType(\PHPExcel_Cell_DataValidation::TYPE_LIST);
$objValidation->setErrorStyle(\PHPExcel_Cell_DataValidation::STYLE_INFORMATION);
$objValidation->setShowErrorMessage(true);
$objValidation->setShowDropDown(true);
$objValidation->setErrorTitle($errorTitle);
$objValidation->setError($errorMessage);
$objValidation->setAllowBlank(false);
$objValidation->setShowInputMessage(true);
$objValidation->setPromptTitle($promptTitle);
$objValidation->setPrompt($prompt);
$objValidation->setFormula1($options);
```

The variable $options is a comma separated string (e.g., '"value 1","value 2","value 3"').

It seems as a bug to me, but is there anything I'm doing wrong?

**Comment by PowerKiKi** [comment]
Could you please give us a **complete** example of code that can work independently and reproduce the issue ? Especially, include the creation of the worksheet and writing it to a file.

Also what version of PHPExcel are you using ?

**Comment by demontiejr** [comment]
Well... I'm basically creating a sheet from a matrix containing my data. Something like this:

```
        $spreadsheet = Excel::create($template['reference'], function($excel) use($variableGroups) {
            $excel->sheet('Questionario', function($sheet) use($variableGroups) {
                $sheet->setAutoSize(true);
                $column = 0;
                $allVars = [];
                $allQuestions = [];
                foreach ($variableGroups as $group => $variables) {
                    $numVariables = count($variables);
                    $sheet->mergeCells($this->getColumnName($column).'1'
                        .':'.
                        $this->getColumnName($column + $numVariables - 1).'1');
                    $sheet->cell($this->getColumnName($column).'1', function($cell) use($group) {
                        $cell->setValue($group);
                    });

                    foreach ($variables as $variable => $details) {
                        $allVars[] = $variable;
                        $allQuestions[] = $details['question'];

                        $this->createValidation($sheet, $this->getColumnName($column), $details);

                        $column++;
                    }
                }

                $sheet->row(2, $allVars);
                $sheet->row(3, $allQuestions);
            });
        });

        return $spreadsheet->download('xlsx');
```

My createValidation method basically decides what's the type of validation and the prompt message and calls the method which actually creates the validation for a thousand cells in the specified column. The method which creates the list validation is the one I sent in the question.

And I'm using PHPExcel 1.8.*.

**Comment by PowerKiKi** [comment]
Unfortunately it doesn't seem to be a [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve) since that code seems to be dependent on Laravel. Try to reproduce it independently from your existing application, in a file that can run by other people without any other  dependencies than PHPExcel itself.

**Comment by demontiejr** [comment]
Well, the first code snippet I sent seems to be independent of Laravel and I'm pretty sure this is the part that is breaking the spreadsheet (since I could generate valid spreadsheets without the validation). Actually, even this part of the code works normally (just the prompt message without any validation):

```
$objValidation = $cell->getDataValidation();
$objValidation->setAllowBlank(false);
$objValidation->setShowInputMessage(true);
$objValidation->setPromptTitle($promptTitle);
$objValidation->setPrompt($prompt);
```

But anyway, unfortunately I won't have time to try to reproduce it without using Laravel. I just thought someone could have a thought about whether I could be doing something wrong.

**Comment by demontiejr** [comment]
Just for the benefit of others, I decided to take another look at this problem today and I found these examples: https://github.com/PHPOffice/PHPExcel/blob/master/Examples/15datavalidation.php#L102.

It says that the hole list of comma-separated values should be between quotes. As I didn't know about it, I was having each individual value between quotes and it was breaking the spreadsheet. The weird thing is that it still opens normally with LibreOffice.
